miscellaneous/etroc1_cal_studies.py

```diff
@@ -0,0 +1,214 @@
+from pathlib import Path # Pathlib documentation, very useful if unfamiliar:
+                         #   https://docs.python.org/3/library/pathlib.html
+
+import lip_pps_run_manager as RM
+
+import logging
+import pandas
+import numpy
+import sqlite3
+import plotly.graph_objects as go
+
+from utilities import apply_event_filter, filter_dataframe
+
+def plot_cal_code_task(
```

Comment:
I would suggest putting this whole function in the utilities.py script so that it can be reused in other scripts. Like this all the plotting logic is also contained inside a single file.

I would also recommend to structure the code in a similar way to the other plot_task functions so that they are modular and reusable as single parts if needed.

Finally, take a look at the replot.py script. You should add some logic there as well so that these plots are remade if the replot script sees that these plots were previously created.